include/deal.II/base/work_stream.h

```diff
@@ -874,14 +1019,352 @@ namespace WorkStream
+                typename Copier,
+                typename Iterator,
+                typename ScratchData,
+                typename CopyData>
+      void
+      run(const std::vector<std::vector<Iterator>> &colored_iterators,
+          Worker                                    worker,
+          Copier                                    copier,
+          const ScratchData &                       sample_scratch_data,
+          const CopyData &                          sample_copy_data,
+          const unsigned int                        chunk_size)
+      {
+        // loop over the various colors of what we're given
+        for (unsigned int color = 0; color < colored_iterators.size(); ++color)
+          if (colored_iterators[color].size() > 0)
+            {
+              using WorkerAndCopier = internal::tbb_colored::
+                WorkerAndCopier<Iterator, ScratchData, CopyData>;
+
+              using RangeType = typename std::vector<Iterator>::const_iterator;
+
+              WorkerAndCopier worker_and_copier(worker,
+                                                copier,
+                                                sample_scratch_data,
+                                                sample_copy_data);
 
+              parallel::internal::parallel_for(
+                colored_iterators[color].begin(),
+                colored_iterators[color].end(),
+                [&worker_and_copier](
+                  const tbb::blocked_range<
+                    typename std::vector<Iterator>::const_iterator> &range) {
+                  worker_and_copier(range);
+                },
+                chunk_size);
+            }
+      }
 
+    }    // namespace tbb_colored
 #  endif // DEAL_II_WITH_TBB
 
 
+
+#  ifdef DEAL_II_WITH_TASKFLOW
+    namespace taskflow_v4
+    {
+      template <typename CopyData>
+      class Chunk
+      {
+      public:
+        Chunk(const unsigned int count, const CopyData &copy_data)
+          : copy_datas(count, copy_data)
+        {}
+        std::vector<CopyData> copy_datas;
+      };
+
+
+
+      /**
+       * WorkStream without colored iterators.
+       *
+       * Work in chunks of size @p chunk_size (only if we have enough items, otherwise 1) and
+       * use a thread-local scratch object
+       */
+      template <typename Worker,
+                typename Copier,
+                typename Iterator,
+                typename ScratchData,
+                typename CopyData>
+      void
+      run(const Iterator &                         begin,
+          const typename identity<Iterator>::type &end,
+          Worker                                   worker,
+          Copier                                   copier,
+          const ScratchData &                      sample_scratch_data,
+          const CopyData &                         sample_copy_data,
+          const unsigned int                       chunk_size = 8)
+      {
+        if (MultithreadInfo::n_threads() == 1)
```

Comment:
Couldn't you make the code simpler here by just calling the version in the `sequential` namespace?